CONTRIBUTING.md

```diff
@@ -51,6 +53,24 @@ following terms in the code, messages, or comments. Note the alternatives we wil
    * controller, host / device, worker, proxy
    * director / performer
 
+#### Git Commits
+
+##### Rules for Git Commit Message Style
+- Separate subject from the body with a blank line
+- Use past tense verbs
+- Wrap the lines at 72 characters
+- Use the body to explain the changes
```

Comment:
I'm slightly torn on including the "why" in the commit body. Generally I focus on describing the change at a high, but complete, level; but the *why* behind something changed in a commit seems like it might be hard to capture without bloating a commit body to something on the order of the Jira task, or some of the comments in code.

Listing exactly what changed is still very valuable when trying to find which commits may or may not have made a given change, without having to bisect or dig through diffs.

Reply:
This is definitely not a strict requirement to have in the git commit. I suppose I would expect this in a minority of changes where they why is really important (because what the code is doing may look very counter-intuitive, so without explanation it would be confusing to the reader). And even then, the why doesn't need to be in the commit message; it could be in a comment in the code. Maybe we can phrase this in a way that is clear? E.g. `In the few cases it makes sense, please include why the change was needed, or why it was implemented in a certain way`?

Reply:
And that would be a separate bullet point from the `Use the body to explain what changed`

Reply:
I think that makes sense! I added `In the few cases it makes sense, please include reason the change was needed, or why it was implemented in a certain way` and a sub-bullet of `Use the body to explain what changed`.